app/src/androidTest/java/com/okta/oidc/example/MfaTest.java

```diff
@@ -0,0 +1,237 @@
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static androidx.test.platform.app.InstrumentationRegistry.getInstrumentation;
+import static com.okta.oidc.example.Utils.EXCHANGE_TIMEOUT;
+import static com.okta.oidc.example.Utils.RECENTS_TIMEOUT;
+import static com.okta.oidc.example.Utils.TEXT_APPROVE;
+import static com.okta.oidc.example.Utils.TEXT_DENY;
+import static com.okta.oidc.example.Utils.TEXT_SEND_PUSH;
+import static com.okta.oidc.example.Utils.NOTIFICATION_TIMEOUT;
+import static com.okta.oidc.example.Utils.TRANSITION_TIMEOUT;
+import static com.okta.oidc.example.Utils.WAIT_FOR_POLL_TIMEOUT;
+import static com.okta.oidc.example.Utils.customTabInteractionWithMfa;
+import static junit.framework.TestCase.assertTrue;
+import static org.hamcrest.core.StringContains.containsString;
+
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public class MfaTest {
+    //app resource ids
+    private static final String ID_GET_PROFILE = "com.okta.oidc.example:id/get_profile";
+    private static final String ID_CLEAR_ALL = "com.android.systemui:id/dismiss_text";
+    private static final String OKTA_OIDC_APP = "Okta OIDC Plain";
+    private static final String OKTA_VERIFY = "com.okta.android.auth";
+    private static final String NEXUS_LAUNCHER = "com.google.android.apps.nexuslauncher";
+    private static final String SYSTEM_UI = "com.android.systemui";
+    private static final String ID_TITLE = "android:id/title";
+    private static final String CLASS_TEXTVIEW = "android.widget.TextView";
+    private static final String NOTIFICATION = "MFA Notification";
+    private static final String DENY_TEXTVIEW = "You have chosen to reject this login.";
+
+    private UiDevice mDevice;
+    @Rule
+    public ActivityTestRule<PlainActivity> activityRule = new ActivityTestRule<>(PlainActivity.class);
+
+    private UiObject getProfileButton() {
+        return mDevice.findObject(new UiSelector().resourceId(ID_GET_PROFILE));
+    }
+
+    private UiObject getSendPushButton() {
+        return mDevice.findObject(new UiSelector().text(TEXT_SEND_PUSH));
+    }
+
+    private UiObject getApproveButton() {
+        return mDevice.findObject(new UiSelector().description(TEXT_APPROVE));
+    }
+
+    private UiObject getDenyButton() {
+        return mDevice.findObject(new UiSelector().description(TEXT_DENY));
+    }
+
+    private void clearAllNotifications() {
+        mDevice.openNotification();
+        mDevice.wait(Until.hasObject(By.pkg(SYSTEM_UI)), TRANSITION_TIMEOUT);
+        UiObject2 clearAll = mDevice.findObject(By.res(ID_CLEAR_ALL));
+        if (clearAll != null) {
+            clearAll.click();
+        } else {
+            mDevice.click(mDevice.getDisplayWidth() / 2, mDevice.getDisplayHeight() / 2);
```

Comment:
why are we just clicking the middle of the screen? Isn't clear all usually top right somewhere? Plus if we can't find it seems like we should just fail the test instead of clicking at random. Otherwise this may cause hard to track down flakiness

Reply:
The center click is to close the notification drawer. If it doesn't have a clearAll button, it should mean that no okta verify notifications are pending so it can proceed with the tests.

Reply:
We should swipe up instead to close the app drawer to be more reliable. If there are multiple notifications for whatever reason then we could end up clicking one by mistake.

Reply:
fixed. Change to swipe.